sdk/lib/drivers/sound/mmixer/wave.c

```diff
@@ -614,6 +614,46 @@ MMixerGetWaveOutCount(
     return MixerList->WaveOutListCount;
 }
 
+MIXER_STATUS
+MMixerGetWavePosition(
+    _In_ PMIXER_CONTEXT MixerContext,
+    _In_ HANDLE PinHandle,
+    _Out_ DWORD * Position)
+{
+    KSAUDIO_POSITION AudioPosition;
+    KSPROPERTY Property;
+    MIXER_STATUS Status;
+    ULONG Length;
+
+    /* verify mixer context */
+    Status = MMixerVerifyContext(MixerContext);
+
+    if (Status != MM_STATUS_SUCCESS)
+    {
+        /* invalid context passed */
+        return Status;
+    }
+
+    Property.Id = KSPROPERTY_AUDIO_POSITION;
+    Property.Set = KSPROPSETID_Audio;
+    Property.Flags = KSPROPERTY_TYPE_GET;
+
+    Status = MixerContext->Control(PinHandle, IOCTL_KS_PROPERTY, &Property, sizeof(KSPROPERTY), &AudioPosition, sizeof(KSAUDIO_POSITION), &Length);
+
+    if (Status == MM_STATUS_SUCCESS)
+    {
+        /* store audio position */
+        *Position = (DWORD)AudioPosition.PlayOffset;
+    }
+    else
+    {
+        /* failed */
+        *Position = 0;
```

Comment:
In principle, something gets returned only in case of success, not on failure necessarily.

Reply:
OK, removed this else case.